client/components/CreateAccount/CreateAnonymousAccount.jsx

```diff
@@ -0,0 +1,179 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import {
+    Button,
+    Header,
+    Loader,
+    Modal,
+    Popup,
+    Segment
+} from 'semantic-ui-react';
+import { logIn } from '@client/actions';
+import Session from '@client/util/session';
+import anonymousUsername from './anonymousUsername';
+import './CreateAnonymousAccount.css';
+
+class CreateAnonymousAccount extends Component {
+    constructor(props) {
+        super(props);
+
+        const from =
+            this.props.location.state && this.props.location.state.from;
+
+        this.state = {
+            from,
+            username: ''
+        };
+
+        this.onClick = this.onClick.bind(this);
+        this.onSubmit = this.onSubmit.bind(this);
+    }
+
+    componentDidMount() {
+        this.generateUnusedAnonymousUsername();
+    }
+
+    async generateUnusedAnonymousUsername() {
+        const username = anonymousUsername();
+        try {
+            await fetch(`/api/auth/signup/usernames/${username}/exists`);
+        } catch (error) {
+            if (error) {
+                return await this.generateUnusedAnonymousUsername();
+            }
+        }
+
+        if (username) {
+            this.setState({ username });
+        }
+    }
+
+    async onSubmit() {
+        const { from, username } = this.state;
+        const body = JSON.stringify({
+            username
+        });
+
+        const { error, message } = await (await fetch('/api/auth/signup', {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body
+        })).json();
+
+        if (error) {
+            this.setState({ message });
+            return;
+        }
+
+        if (username) {
+            Session.create({ username, timeout: Date.now() });
+            // Step outside of react to force a real reload
+            // after signup and session create
+            location.href = from ? from.pathname : '/';
+        }
+    }
+
+    onClick(event, { name }) {
+        if (name === 'no') {
+            this.generateUnusedAnonymousUsername();
+        }
+        if (name === 'nvm') {
+            location.href = '/';
+        }
+    }
+
+    render() {
+        const { username } = this.state;
+        const { onClick, onSubmit } = this;
+        const header = username
+            ? 'Do you like this temporary user name?'
```

Comment:
I guess what makes this account "temporary" is that it doesn't have a password associated with it?